src/session.ts

```diff
@@ -32,14 +35,27 @@ async function updateSession(request: NextRequest, debug: boolean) {
 
   newRequestHeaders.delete(sessionHeaderName);
 
+  const matchedPaths: string[] = middlewareAuth.unauthenticatedPaths.filter((route) => {
+    const routeRegexStr: string = parseRoute(route);
+    const regex: RegExp = new RegExp(routeRegexStr);
+
+    return regex.exec(request.nextUrl.pathname);
+  });
+
+  // If the user is logged out and this path isn't on the allowlist for logged out paths, redirect to AuthKit.
+  if (middlewareAuth.enabled && matchedPaths.length === 0 && !session) {
+    if (debug) console.log('Unauthenticated user on protected route, redirecting to AuthKit');
+    return NextResponse.redirect(await getAuthorizationUrl({ returnPathname: new URL(request.url).pathname }));
+  }
+
   // If no session, just continue
   if (!session) {
     return NextResponse.next({
       request: { headers: newRequestHeaders },
     });
   }
 
-  const hasValidSession = await verifyAccessToken(session.accessToken);
+  const hasValidSession: boolean = await verifyAccessToken(session.accessToken);
```

Comment:
Why did we need to add a type annotation here? Isn't this inferred anyway?

Reply:
Oops, didn't mean to commit this.